capsules/src/nonvolatile_storage_driver.rs

```diff
@@ -526,46 +586,54 @@ impl Driver for NonvolatileStorage<'_> {
     /// - `1`: Return the number of bytes available to userspace.
     /// - `2`: Start a read from the nonvolatile storage.
     /// - `3`: Start a write to the nonvolatile_storage.
-    fn command(&self, arg0: usize, arg1: usize, _: usize, appid: AppId) -> ReturnCode {
-        let command_num = arg0 & 0xFF;
-
+    fn command(
+        &self,
+        command_num: usize,
+        offset: usize,
+        length: usize,
+        appid: AppId,
+    ) -> CommandReturn {
         match command_num {
-            0 =>
-            /* This driver exists. */
-            {
-                ReturnCode::SUCCESS
+            0 /* This driver exists. */ => {
+                CommandReturn::success()
             }
 
-            // How many bytes are accessible from userspace.
-            1 => ReturnCode::SuccessWithValue {
-                value: self.userspace_length,
+            1 /* How many bytes are accessible from userspace */ => {
+                // TODO: Would break on 64-bit platforms
```

Comment:
We have 64-bit return codes (now? compared to when this was first ported?), should that be used here?

Reply:
We do not currently. But the internal Rust APIs for system calls were designed to hopefully be able to support 64-bit platforms (note the use of usize). We currently only have an ABI defined for 32 bit platforms.